### PR TITLE
Feature handle new local files

### PR DESCRIPTION
**Pull request by ThunderKey**
Tested with:
* Removing the cache file: `rm ~/.local/share/kitovu/filecache.json`
* Changing a file
* `kitovu sync`
* Everything has the state `FileState.NO_CHANGES` and does nothing (but update teh filecache)
* One Exception is the edited file which has `FileState.BOTH_CHANGED` and gets downloaded

Maybe we could make the `filecache.File` class wrapping `local_digest` and `remote_digest`. We then could move the `_compare_digests` to that class as well instead of having the `FileCache` do everything

### CODE REVIEW

File: src/kitovu/sync/filecache.py [hunk 1]
```diff
@@ -143,6 +143,8 @@ def discover_changes(self,
         if not local_full_path.exists():
             return FileState.NEW
 
+        if local_full_path not in self._data:
+            self._data[local_full_path] = File(cached_digest='', plugin_name=plugin.NAME)
```
Comment (The-Compiler) [inline src/kitovu/sync/filecache.py]: mypy fails with `Argument "plugin_name" to "File" has incompatible type "Optional[str]"; expected "str"` here, but an `assert plugin.NAME is not None` above (probably fine to do it even outside of the `if`) will help.

Comment (The-Compiler) [reply src/kitovu/sync/filecache.py]: Not sure if it's a good idea to use `''` for the cached digest, that makes it look like it's a bug in digest calculation when you don't know about it. What about using `None` instead?

Comment (The-Compiler) [reply src/kitovu/sync/filecache.py]: Also, please add a newline after this line, as it's unrelated to the rest of the code creating `file` (makes it easier to scan, IMHO).

File: src/kitovu/sync/filecache.py [hunk 2]
```diff
@@ -152,4 +154,8 @@ def discover_changes(self,
 
         remote_digest: str = plugin.create_remote_digest(remote_full_path)
         local_digest: str = plugin.create_local_digest(local_full_path)
+
+        # Case 5 if both change to the same values
+        if remote_digest == local_digest and file.cached_digest != remote_digest:
```
Comment (The-Compiler) [inline src/kitovu/sync/filecache.py]: I'm not sure if this is the correct approach, but it's too hot for me to think properly right now :sweat_smile:
Need to think a bit more about it - wouldn't it be better to only overwrite the cached digest if we know we don't have any (i.e. it's `''` or `None`)?

@bytinbit maybe you can take a look too?

**Review by The-Compiler — CHANGES_REQUESTED** [review]

File: src/kitovu/sync/filecache.py (hunk 1 above)
Comment (ThunderKey) [reply src/kitovu/sync/filecache.py]: Good idea with setting the cached_digest to `None`. That way it's clear if it from the plugin or not (beacause plugins can only return strings). Additionally I've added a `assert cached_digest is not None` in the `to_dict` method to ensure it won't get saved that way (either download => local_digest or set it to the current if the `remote_digest` and `local_digest` are equal

File: src/kitovu/sync/filecache.py (hunk 2 above)
Comment (The-Compiler) [reply src/kitovu/sync/filecache.py]: The comment should make this situation clear somehow:

image

**Comment by The-Compiler** [comment]
LGTM now!

**Review by The-Compiler — APPROVED** [review]

**Comment by The-Compiler** [comment]
Oh, something I forgot: It'd be nice to add a test for this...

**Comment by ThunderKey** [comment]
Added tests in #35

**Pull request merged**